### PR TITLE
Offline activation code

### PR DESCRIPTION
**Pull request by DmitriyNoa**

### CODE REVIEW

File: src/UserService.ts [hunk 1]
```diff
@@ -371,6 +386,20 @@ class UserService extends Base {
     return (json as unknown) as SubscriptionInfo;
   }
 
+  async getUserSubscriptionOfflineActivationCode(username: string, subscriptionId: string, offlineCodeActivationData: ActivationCodeData): Promise<OfflineActivationCode> {
+    const { apiEndpointAddress, fetch } = this.lensPlatformClient;
+    const url = `${apiEndpointAddress}/users/${username}/subscription-seats/${subscriptionId}/activation-code?accessToken=a${offlineCodeActivationData.accessToken}&refreshToken=${offlineCodeActivationData.refreshToken}&idTokenTest=${offlineCodeActivationData.idToken}`;
```
Comment (panuhorsmalahti) [inline src/UserService.ts:391]: It's unsafe to pass sensitive data in url. (Yeah, need to change the backend implementation really).

Comment (DmitriyNoa) [reply src/UserService.ts:391]: We can keep activation code on frontend then as it's generation requires token data which we can't get on backend

Comment (DmitriyNoa) [reply src/UserService.ts:391]: Not sure if passing in request headers will be much different

Comment (panuhorsmalahti) [reply src/UserService.ts:391]: Passing in headers is considered safe (that's how Authorization header is passed normally, for example). But yeah, generating it online (like currently do AFAIK) is also fine.

Comment (nevalla) [reply src/UserService.ts:391]: Yes we can keep the code generation on client at least for now.

Comment (chenhunghan) [reply src/UserService.ts:391]: Headers are encrypted if it's via HTTPs (afaik)

Comment (DmitriyNoa) [reply src/UserService.ts:391]: Moved to a single POST and post body

File: src/UserService.ts [hunk 2]
```diff
@@ -425,6 +440,32 @@ class UserService extends Base {
...
+        404(error) {
+          const message = error?.body.message;
+
+          if (typeof message === "string") {
+            if (message.includes("User")) {
+              return new UserNameNotFoundException(username);
+            }
+          }
+
+          return new NotFoundException("Subscription seat subscriptionSeatId not found");
```
Comment (panuhorsmalahti) [inline src/UserService.ts:458]: Do we want to say `subscriptionSeatId` in the error message? We could use the actual `subscriptionSeatId` value instead.

**Review by panuhorsmalahti — APPROVED** [review]

**Pull request merged**